project/stac_api/models.py

```diff
@@ -1,28 +1,307 @@
+        default=[
+            'eo',
+            'proj',
+            'view',
+            'https://data.geo.admin.ch/stac/geoadmin-extension/1.0/schema.json'
+        ],
+        editable=False
+    )
+
+    stac_version = models.CharField(blank=False)
+    GeoJSON_type = models.CharField(default="Feature", blank=False, editable=False)
+    # specifies the GeoJSON type and MUST be set to "Feature".
+    assets = models.TextField()
+    # this is defined as required here: https://github.com/radiantearth/stac-spec/blob/v0.9.0/item-spec/item-spec.md
+    # and is meant to contain a dictionary of asset objects than can be downloaded, each with a unique key
+    # will be auto-populated on every update of an asset inside this item.
+    # TODO: overwrite assets save() function accordingly.
+    location = models.URLField()
+
+    def __str__(self):
+        return self.item_name
+
+    def clean(self):
+        if len(self.SW) != len(self.NE):
+            raise ValidationError(_(
+                'If intended spatial extent is 3D, then elevation needs to be defined in southwesterly most and northeasterly most point.'
+            ))
+        elif (len(self.SW) != 4) and (len(self.SW) != 6):
+            raise ValidationError(_('Bounding box incorrectly defined.'))
+
+    def save(self, *args, **kwargs):
+        #TODO: on every item update:
+        # 1. re-calculate the envelope over all bboxes of all items inside this collection
+        #    and update the collections' bbox accordingly
+        #
+        # 2. loop over all items inside the collection and set the "start_date" of
+        #    the collection to the oldest date found in item dates. Same for
+        #    collections' end_date (set to most recent date found in items)
+        #
+        # 3. also loop over all items inside the collection and update the
+        #    collections' "summaries"-values (not sure if this should be done on
+        #    asset level instead. I guess so, as "proj" will be defined on asset
+        #     level, whereas gsd is defined on item and asset level...)
+        #
+        # 4.  loop over all items and update the collections "provider" property
+        #     on every item update
+
+        super().save(*args, **kwargs)
+
+
+class Asset(models.Model):
+    feature = models.ForeignKey(Item, on_delete=models.CASCADE)
+    collection = models.ForeignKey(Collection, on_delete=models.CASCADE, blank=True, editable=False)
+    id = models.BigAutoField(primary_key=True)
+
+    # using BigIntegerField as primary_key to deal with the expected large number of assets.
+
+    # alter save-function, so that the corresponding collection of the parent item of the asset
+    # is saved, too.
+    def save(self, *args, **kwargs):
+        self.collection = self.feature.collection
+        super().save(*args, **kwargs)
+
+    # using "_name" instead of "_id", as "_id" has a default meaning in django
+    asset_name = models.CharField(unique=True, blank=False)
```

Comment:
you'll have to define `max_length` for char fields

Reply:
done